generator/internal/genclient/language/internal/rust/rust_test.go

```diff
@@ -86,3 +86,70 @@ func TestToPascal(t *testing.T) {
 		}
 	}
 }
+
+func TestMessageNames(t *testing.T) {
+	message := &genclient.Message{
+		Name: "Replication",
+		ID:   "..Replication",
+		Fields: []*genclient.Field{
+			{
+				Name:     "automatic",
+				Typez:    genclient.MESSAGE_TYPE,
+				TypezID:  "..Automatic",
+				Optional: true,
+				Repeated: false,
+			},
+		},
+	}
+	nested := &genclient.Message{
+		Name: "Automatic",
+		ID:   "..Replication.Automatic",
+	}
+
+	api := genclient.NewTestAPI([]*genclient.Message{message, nested}, []*genclient.Enum{}, []*genclient.Service{})
+
+	c := &Codec{}
+	if got := c.MessageName(message, api.State); got != "Replication" {
+		t.Errorf("mismatched message name, want=Replication, got=%s", got)
```

Comment:
nit: goes convention for testing is backwards from most languages. I would usually expect these strings to be written as "got Replication, want %s"

Reply:
Fixed.